titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/common/DifferenceResolverUtils.java

```diff
@@ -239,7 +238,7 @@ public static int countActiveNotStartedTasks(EntityHolder refJobHolder, EntityHo
         Set<String> unassignedIpAddressIds = refJobView.getJob().getJobDescriptor().getContainer().getContainerResources().getSignedIpAddressAllocations()
                 .stream()
                 .map(signedIpAddressAllocation -> signedIpAddressAllocation.getIpAddressAllocation().getAllocationId())
-                .collect(Collectors.toCollection(ConcurrentSkipListSet::new));
+                .collect(Collectors.toCollection(HashSet::new));
```

Comment:
now that this is not concurrent, I believe you also need to change the lines below to stop using the `Stream` API, and do a regular plain old for loop to remove items from the `Set`.

`Stream#forEach` requires synchronization and thread safety when having side effects:

```
     * <p>The behavior of this operation is explicitly nondeterministic.
     * For parallel stream pipelines, this operation does <em>not</em>
     * guarantee to respect the encounter order of the stream, as doing so
     * would sacrifice the benefit of parallelism.  For any given element, the
     * action may be performed at whatever time and in whatever thread the
     * library chooses.  If the action accesses shared state, it is
     * responsible for providing the required synchronization.
```

Not a big deal in practice since we know the stream will not be parallel, but it's better to avoid it for correctness and to prevent surprises in the future.

Reply:
Fixed.